### DISCUSSION TITLE
Arcade Mode

### DISCUSSION THREAD

**Discussion by ghost** [ppy/osu#13071] [Ideas]
## Arcade mode
Basically, a "kiosk-esque" mode for osu!lazer that makes it more suitable for deployment at LAN parties and similar events. From my own experience, there needs to be a way to prevent any random passerby from messing around with settings and beatmaps, among other things. (This is targeted towards touchscreen-based environments where we can assume that the keyboard will be locked away)

## How it could work
On startup, the game will look for a configuration file, say, `arcade.json` located in the root directory. This will activate arcade mode and enable one to customize the arcade experience. Options may include:

* Disabling exit buttons
* Restricting game-modes to a predefined list
* Going straight into the beatmap selection screen (single player mode), bypassing the main menu
* Hiding top-bar buttons like the changelog, chat, or osu!direct
* Disabling user login and preventing the dialog from popping up automatically
* Password-protecting the settings, options, and notification/tasks menus (they can be hidden by default, then shown after the osu! logo is clicked/tapped for `x` times, at which point the password prompt pops up)
* Using a local scoreboard with a prompt asking for a name at the end of each game

Additionally, inspired by the original osu!arcade, this mode may also enable connectivity to external hardware control devices, such as lighting controllers through a serial connection. (I don't know if this will require an engine change though...)

I *might* be able to help with the programming for this feature, but I'd probably need some help as C# isn't exactly what I do everyday...

**Comment by bdach** [comment]
Since the game is constructed with components, it might be actually preferable to prepare a separate version of the client reusing all of the necessary parts instead of peppering the existing code with flag checks absolutely everywhere. Nevertheless, this is probably very low in priority, if still sort of interesting.
